virttest.utils_misc: Remove the non-woring decode

ldoktor opened the pull request:
The 9043d263c7bd935474191dcf38fa5a24f60ce4a8 breaks the unicode support
again. Before that commit both, unicode() and str() were supported, when
they followed the defined encoding, anyway I can see that probably
someone tries to log already encoded unicode, which might fail. So let's
remove all those different handlings and use the `astring`
implementation of the lenient encode, which first assumes the right
encoding and when that fails it assumes it is already encoded string.

Please don't break this function anymore as it is crucial for
unattended_install and breaks the CI!

Signed-off-by: Lukáš Doktor <ldoktor@redhat.com>

xutian commented:
@ldoktor  before 9043d26  rhel5.11 and rhel6.9 installation faild because unicode decode exception,  details see  rh bug#1436501 and our CI test also blocked. 
And thanks your fix, let's have a try with 5.11 and 6.9 RHEL guest, results will post later.

Thanks,
Xu

xutian commented on virttest/utils_misc.py:
```diff
@@ -477,8 +478,7 @@ def log_line(filename, line):
         timestr = time.strftime("%Y-%m-%d %H:%M:%S")
-        line = unicode("%s: %s" % (timestr, line), errors="ignore")
-        line = line.encode(ENCODING, errors='ignore')
+        line = string_safe_encode(line)
```
time string lost here.

clebergnu replied:
True.

sathnaga replied:
ENCODING = os.environ.get("PYTHONENCODING", "utf-8") this line(52) can be removed, 

line = string_safe_encode(line) --> line = string_safe_encode("%s: %s" % (timestr,line))

ldoktor replied:
Oups, I'm sorry about the missing time, fix in the next version along with removing the unnecessary ENCODING line.

yanan-fu commented:
@ldoktor @xutian  
With this fix,  rhel 7 guest is ok,  but cause the bz1436501 that fixed by 9043d263c7bd935474191dcf38fa5a24f60ce4a8 again.

 (1/3) smp_8.8192m.repeat1.Host_RHEL.m7.u4.spice.qcow2.virtio_scsi.up.virtio_net.RHEL.7.4.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads: PASS (1270.31 s)
 (2/3) smp_8.8192m.repeat1.Host_RHEL.m7.u4.spice.qcow2.virtio_blk.up.virtio_net.RHEL.5.11.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads: ERROR (2341.09 s)
 (3/3) smp_8.8192m.repeat1.Host_RHEL.m7.u4.spice.qcow2.virtio_blk.up.virtio_net.RHEL.6.9.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads: ERROR (2640.45 s)

ldoktor commented:
Hello guys, basically this all started with @Andrei-Stepanov who had a problem installing RHEL.6 as it produces `\xff` character in serial console during boot. The old aexpect didn't care about this so it was a `str("\xff")`, which can be stored directly. Anyway the new aexpect (python-3 compatible) has to correctly specify the encoding and the `"\xff"` is an invalid character. It's properly detected and stored as `u"\xff"`.

Now the problem is that pure `open(file).write(u"\xff")` fails as `open` opens the file assuming `ascii` encoding. The `"\xff"` can be written as it's just a bitstream, but `u"\xff"` is a special character, which is not present in `ascii`.

The original fix allowed writing unicode and string characters, but it did not allow writing encoded `"\xff"` characters. Then @xutian changed it to always write `unicode`, but the use of `"%s"` makes it impossible to actually pass `unicode` as an argument.

This fix should always work as it uses a lenient implementation of `string_safe_encode` which assumes it's encoded unicode and when it fails it assumes it's unicode and decodes it.

There is one possible issue when it's not a proper unicode and I'm addressing that in https://github.com/avocado-framework/avocado/pull/1901

Anyway I did this modification in a hurry so after the meetings I'll double-check it works well...

xutian commented:
@ldoktor do you have any idea to make it works both rhel6 and rhel7 guest. thanks.

ldoktor commented:
Hard to say, I'd need more info. I believe that this version should work well when we include the https://github.com/avocado-framework/avocado/pull/1901 (in both, master and lts), or I can actually add the `try/except` here as well to allow using avocado versions without that fix. Also I need to update this to include the time, I did this fix really ad-hock to make my tests pass.

Andrei-Stepanov commented:
We depend on LTS avocado. Could https://github.com/avocado-framework/avocado/pull/1901 be back-ported to LTS?

ldoktor commented:
@xutian I fixed the missing time issue and added a workaround in case someone is using avocado without the https://github.com/avocado-framework/avocado/pull/1901 (which is still not merged so it makes sense to include it here as well, even if just temporarily).

This version should work well, I tried various different inputs:

```python
>>> from virttest.utils_misc import log_line as a
>>> a("/tmp/aaa", u"šěřščř\xff")
>>> a("/tmp/aaa", "šěřščř\xff")
```
and the result is:
```
$ cat /tmp/aaa
2017-04-05 16:48:23: šěřščřÿ
2017-04-05 16:48:27: šěřščř
```
